src/Libraries/PythonNodeModels/PythonNode.cs

```diff
@@ -33,6 +33,7 @@ public PythonCodeMigrationEventArgs(string oldCode, string newCode)
     public abstract class PythonNodeBase : VariableInputNode
     {
         private PythonEngineVersion engine = PythonEngineVersion.Unspecified;
+        private PythonEngineVersion cached_engine = PythonEngineVersion.Unspecified;
```

Comment:
I think the consensus around C# naming convention these days is do not user underscore at all, would you try `cachedEngine`?